src/sdl_font.cpp

```diff
@@ -279,6 +279,29 @@ CachedTTFFont::CachedTTFFont(
         throw std::runtime_error( TTF_GetError() );
     }
     TTF_SetFontStyle( font.get(), TTF_STYLE_NORMAL );
+
+    ImGuiIO &io = ImGui::GetIO();
+    static const std::array<ImWchar, 17> ranges = {
+        0x0020, 0x052F,
+        0x1D00, 0x1DFF,
+        0x2000, 0x206F,
+        0x20A0, 0x20CF,
+        0x2100, 0x214F,
+        0x2190, 0x22FF,
+        0x2500, 0x27BF,
+        0xC900, 0xC9FF,
+        //0x0020, 0xCFFF,
+        0
+    };
+    imgui_font = io.Fonts->AddFontFromFileTTF( typeface.c_str(), fontsize, nullptr, ranges.data() );
+    if( imgui_font && !io.FontDefault && typeface.find( "unifont", 0 ) != std::string::npos ) {
+        io.FontDefault = imgui_font;
```

Comment:
ImGui DOES support font fallback but not in the way you'd think - you just have to tell it what glyphs to use for a certain font and it caches those glyphs in memory. If you want to only use certain glyphs of a font that's perfectly fine. From what I understand, though, we need to figure all of it out ahead of time.

Reply:
Been chipping away at this one. I discovered that ImGui supports loading fonts in "merge mode" which means loading with a fallback. It doesn't quite work like you'd expect, though - you need to know ahead of time what glyph ranges you want to load for a given font. Maybe the supported ranges of each font can be read and passed into ImGui?

Reply:
So I'm still having some difficulty with this. I learned from @andrei8l that on packaged Linux installs, some of the code for "cleaning up" the file path of the typeface within the CachedTTFFont constructor is needed before i can pass the path to ImGui. I will probably resolve this story by only loading the first instance of unifont into ImGui. In order to truly support fallback we need to dynamically load glyphs into ImGui based on which TTF files support which glyphs.

I can't really move the io.FontDefault assignment to where you want, because I can only set it for TTF fonts. Maybe I can use ensure_unifont_loaded and hard-code unifont as the font for ImGui for now?